Scala/Scala.sublime-syntax

```diff
@@ -89,7 +90,7 @@ contexts:
       captures:
         1: keyword.declaration.scala
         2: entity.name.function.declaration
-    - match: '(case)?\b(class|trait|object)\s+([^\s\{\(\[]+)'
+    - match: '\b(case\s+)?(class|trait|object)\s+([^\s\{\(\[]+)'
```

Comment:
Non-capturing groups do not introduce backtracking (that would be terrible!). You might be thinking of atomic groups.